lib/processor/upload.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const path = require('path');
+const fs = require('fs');
+const Q = require('q');
+const s3 = new (require('aws-sdk')).S3();
+const COPY_VARIATIONS = ['broadcast'];
+
+// upload mp3 and make copies as needed
+module.exports = (file) => {
+  let dest = `${file.path}/${file.name}`;
+  let type = file.mimeType();
+
+  let uploadOriginal;
+  if (file.s3Bucket && file.s3Key) {
+    uploadOriginal = s3Copy(file.s3Bucket, file.s3Key, dest, type);
+  } else {
+    uploadOriginal = localUpload(file.localPath, dest, type);
+  }
+
+  let fileParts = path.parse(file.name);
+  let fileCopies = COPY_VARIATIONS.map(vary => {
```

Comment:
If this part of cms was a bit smarter, could avoid needing to make `_broadcast` copy of the file:
https://github.com/PRX/cms.prx.org/blob/master/app/representers/api/audio_file_representer.rb#L29